### PR TITLE
build: remove klaw dependency

### PR DESCRIPTION
**Pull request by ckerr**
#### Description of Change

A side refactor I found while working on something nearby.

We don't need a klaw devdep anymore, since Node 20 has a recursive option for `fs.readdir()`.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Review by codebytere — APPROVED** [review]

**Review by samuelmaddock — APPROVED** [review]
nice

**Review by jkleinsc — CHANGES_REQUESTED** [review]

File: script/lib/utils.js [hunk 1]
```diff
@@ -130,18 +129,10 @@ function chunkFilenames (filenames, offset = 0) {
...
-      .on('data', item => {
-        if (test(item.path)) {
-          matches.push(item.path);
-        }
-      });
-  });
+  return fs.readdirSync(top, { encoding: 'utf8', recursive: true })
+    .filter((filename) => path.basename(filename) !== '.bin')
+    .filter((filename) => test(filename))
+    .map((filename) => path.join(top, filename));
```
[the remark this answers is not in the document]
Comment (ckerr) [reply script/lib/utils.js]: fixed in 1b96256e258ef3ac86708a983897af7a2a9f5fad

**Review by jkleinsc — APPROVED** [review]

**Review by jkleinsc — APPROVED** [review]

**Comment by ckerr** [comment]
Backporting this after all, since it will make the fs-extra removal troppable

**Pull request merged**